test/org/sagebionetworks/bridge/services/AuthenticationServiceMockTest.java

```diff
@@ -0,0 +1,284 @@
+        doReturn(STUDY_ID).when(study).getIdentifier();
+        doReturn(true).when(study).isEmailSignInEnabled();
+        doReturn(new StudyIdentifierImpl(STUDY_ID)).when(study).getStudyIdentifier();
+        doReturn(new EmailTemplate("subject","body",MimeType.TEXT)).when(study).getEmailSignInTemplate();
+        doReturn("sender@sender.com").when(study).getSupportEmail();
+        doReturn("Sender").when(study).getName();
+    }
+    
+    @Test
+    public void requestEmailSignIn() throws Exception {
+        doReturn(account).when(accountDao).getAccountWithEmail(study, RECIPIENT_EMAIL);
+        
+        service.requestEmailSignIn(SIGN_IN_REQUEST);
+        
+        verify(cacheProvider).getString(stringCaptor.capture());
+        assertEquals(CACHE_KEY, stringCaptor.getValue());
+        
+        verify(accountDao).getAccountWithEmail(study, RECIPIENT_EMAIL);
+        
+        verify(cacheProvider).setString(eq(CACHE_KEY), stringCaptor.capture(), eq(60));
+        assertNotNull(stringCaptor.getValue());
+
+        verify(sendMailService).sendEmail(providerCaptor.capture());
+        
+        EmailSignInEmailProvider provider = providerCaptor.getValue();
+        MimeTypeEmail email = provider.getMimeTypeEmail();
+        assertEquals(RECIPIENT_EMAIL, email.getRecipientAddresses().get(0));
+        assertEquals("\"Sender\" <sender@sender.com>", email.getSenderAddress());
+        assertEquals("body", email.getMessageParts().get(0).getContent());
+    }
+    
+    @Test(expected = UnauthorizedException.class)
+    public void requestEmailSignInDisabled() {
+        doReturn(false).when(study).isEmailSignInEnabled();
+        
+        service.requestEmailSignIn(SIGN_IN_REQUEST);
+    }
+    
+    @Test(expected = LimitExceededException.class)
+    public void requestEmailSignInLimitExceeded() {
+        doReturn("something").when(cacheProvider).getString(CACHE_KEY);
+        
+        service.requestEmailSignIn(SIGN_IN_REQUEST);
+    }
+    
+    @Test
+    public void requestEmailSignInEmailNotRegistered() {
+        doReturn(null).when(accountDao).getAccountWithEmail(study, RECIPIENT_EMAIL);
+        
+        service.requestEmailSignIn(SIGN_IN_REQUEST);
+
+        verify(cacheProvider, never()).setString(eq(CACHE_KEY), any(), eq(60));
+        verify(sendMailService, never()).sendEmail(any());
+    }
+
+    @Test
+    public void emailSignIn() {
+        doReturn(TOKEN).when(cacheProvider).getString(CACHE_KEY);
+        doReturn(account).when(accountDao).getAccountWithEmail(study, RECIPIENT_EMAIL);
+        doReturn(PARTICIPANT).when(participantService).getParticipant(study, account, false);
+        doReturn(CONSENTED_STATUS_MAP).when(consentService).getConsentStatuses(any());
+        
+        UserSession retSession = service.emailSignIn(CONTEXT, SIGN_IN);
+        
+        assertNotNull(retSession);
+        verify(accountDao).getAccountWithEmail(study, RECIPIENT_EMAIL);
+        verify(cacheProvider).removeString(CACHE_KEY);
```

Comment:
Yes, you're right. added a test that it's not called in a sign in without a password, and that it is called when a password is provided.